src/intrinsics/fb-www/fb-mocks.js

```diff
@@ -91,37 +111,31 @@ function createBabelHelpers(realm: Realm, global: ObjectValue | AbstractObjectVa
       invariant(obj instanceof ObjectValue || obj instanceof AbstractObjectValue || obj instanceof AbstractValue);
       invariant(keys instanceof ArrayValue);
       if (obj.isPartialObject() || obj instanceof AbstractObjectValue || obj instanceof AbstractValue) {
```

Comment:
You could make it even more specific. You only need to go down this path if `obj.mightBeObject() && ((obj instanceof AbstractValue && obj.values.isTop()) || obj.isPartial())`. Primitive values always resolve to empty objects. Abstract objects with known templates have known possible keys. Some of them might be empty but they can become conditional.